Add border radius and padding configuration options for hints

elenapan opened the pull request:
This pull request adds the ability to round the corners of hints with the configuration option `hints.radius`. The default value is set to `0px`.

I was not sure if I should make the corner radius an integer or a string that represents the CSS value. I went with the second, but let me know if you like it otherwise and I will fix the PR.

Known issues:
- Rounding stops working (it is no longer applied) with values higher than 14px (with border width 0). Looks like this is due to the small size of the hint widgets, since when I increase the border width, I can also increase the border radius.
- It does not work with CSS values such as `50%`. This could be related to the issue above or it could be a Qt limitation since Qt only implements a subset of CSS. A value of `50%` is supposed to turn the hint into a circle / ellipse.

By the way, thank you so much for working on this amazing browser <3

arza-zara commented:
Does anything else than single pixel value work? `:set hints.radius "8px 1px"` doesn't. Integer value works and should probably be the default/suggested, consistent with `keyhint.radius` and `prompt.radius`.

elenapan commented:
It does work with low values of `%`, for example `:set hints.radius "10%"`
but visually there is no difference with setting a `10px` value.

Looks like you are right, there is no point in using the CSS value here. I will change it to an integer.

Concerning the roundness not being applied with high border radius values, it seems that the width and height of the hint need to be at least double the border radius for it to work.
Also, I managed to pick a value where some small hint boxes were not rounded at all while some larger ones were.

Here are some suggested solutions:
 - Hardcoding `min-height` and `min-width` of the hint widgets equal to `hints.radius * 2`.

   This guarantees that the roundness is applied on every hint box. However, the hints get too big when a big radius is set. Also it is not intuitive that the size of the hints grows when border radius grows.

- Expose `min-height` and `min-width` or even `padding` to the config.

   This way, the user is free to increase their values manually if they want the roundness to be applied consistently.

   If we go this route, I suggest exposing `padding` for the following reasons:
   - It can accept multi values like `"2px 10px 4px 0"`, allowing for more granular customization while only exposing 1 extra configuration option.
   - With symmetrical values for horizontal padding, e.g. `"4px 0"` which adds 4 pixels of padding left and right, it is possible to center-align the hint text, something which is not possible with `min-width` and `text-align: center`. The latter does not seem to be respected.

Let me know which solution sounds better.

The-Compiler commented:
Agreed! There's the `Padding` config type for that purpose. I like the idea of also exposing `padding`, I vaguely remember someone asking about that kind of thing before.

elenapan commented:
Hi there, it has been a hectic month for me, so I am apologizing for not completing this sooner.

I have changed the `hints.padding` type from string to `Padding`.

Also I changed the title of the pull request to better reflect the changes.

arza-zara commented on qutebrowser/config/configdata.yml:
```diff
@@ -1091,6 +1091,20 @@ hints.border:
+
+hints.radius:
+  default: 0
+  type: Int
```
```suggestion
  type:
    name: Int
    minval: 0
```
[Other radii prevent negative values](https://github.com/qutebrowser/qutebrowser/blob/df45e7401ee76a5243a260179a32e0a11c330a9b/qutebrowser/config/configdata.yml#L1361L1363).

The-Compiler commented:
Apologies for the delay from my side as well - thanks @elenapan for the PR and @arza-zara for the review!

I decided to enable some slight border radius by default as it looks quite nice IMHO:

- c57daa663 Update changelog
- 2435518ac Add some hint padding by default

If I'm not mistaken, you edited `settings.asciidoc` by hand? Note that that's not necessary, as it's autogenerated - see the top of the file for instructions. No biggie, though!
